code/modules/mob/living/simple_animal/hostile/regalrat.dm

```diff
@@ -82,16 +84,62 @@
 	else if(user != src && istype(user,/mob/living/simple_animal/hostile/regalrat))
 		. += "<span class='warning'>Who is this foolish false king? This will not stand!</span>"
 
+/mob/living/simple_animal/hostile/regalrat/handle_environment(datum/gas_mixture/environment)
+	. = ..()
+	if(stat == DEAD || !environment || !environment.gases[/datum/gas/miasma])
+		return
+	var/miasma_percentage = environment.gases[/datum/gas/miasma][MOLES] / environment.total_moles()
```

Comment:
What happens if you're in an environment with no miasma? I think you have to `ASSERT_GASES` before this.

Reply:
Didn't notice the check above, nevermind.